### ISSUE TITLE
Generate an error if trying to use @final with TypedDict

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#7849]
`@final` doesn't work as expected with TypedDicts, but mypy doesn't complain about it:

```py
from typing_extensions import final, TypedDict

@final  # No error here
class C(TypedDict):
    x: float
    y: int
    z: str

class D(C):  # No error here
    pass

```

This is confusing, since users would expect that `@final` prevents structural subtyping, but that's not the case. The simplest fix would be to generate an error if `@final` is used in a TypedDict definition.

Hints: This should be a pretty simple thing to fix.  Look at what happens in `analyze_class` in `mypy/semanal.py`.

Originally reported in #7845.

**Comment by corona10** [comment]
@JukkaL 
Hi, Can I take look ahead with this issue? 
I am not familiar with mypy codebase but this issue looks good for me to be familiar with.

**Comment by TH3CHARLie** [comment]
@JukkaL I have submitted a PR #7865 , please take a look
